maestro-client/src/main/java/maestro/drivers/AndroidDriver.kt

```diff
@@ -159,12 +159,12 @@ class AndroidDriver(
                 }
                 shell("am start-activity -n $appId/${activity.name}")
             } else {
-                shell("monkey --pct-syskeys 0 -p $appId 1")
+                shell("monkey --pct-syskeys 0 -p $appId -c android.intent.category.LAUNCHER 1")
             }
         } catch (ioException: IOException) {
-            shell("monkey --pct-syskeys 0 -p $appId 1")
+            shell("monkey --pct-syskeys 0 -p $appId -c android.intent.category.LAUNCHER 1")
         } catch (saxException: SAXException) {
-            shell("monkey --pct-syskeys 0 -p $appId 1")
+            shell("monkey --pct-syskeys 0 -p $appId -c android.intent.category.LAUNCHER 1")
```

Comment:
Hey, @alishari thanks for your contribution. To ensure, if I am not wrong this selects activity with the launcher category, right?

The `LeakLauncherActivity` also has launcher category which means it could still launch leak canary activity.

```
<activity-alias android:name="leakcanary.internal.activity.LeakLauncherActivity" ... > 
   <intent-filter>
                <action
                    android:name="android.intent.action.MAIN" />
                <category
                    android:name="android.intent.category.LAUNCHER" />
                <category
                    android:name="android.intent.category.LEANBACK_LAUNCHER" />
            </intent-filter>
</.activity-alias>
```

Reply:
Yes, you are right, I will look into it to find a solution.